packages/@aws-cdk/aws-ecs/lib/log-drivers/generic-log-driver.ts

```diff
@@ -0,0 +1,47 @@
+import { Construct } from '@aws-cdk/core';
+import { ContainerDefinition } from '../container-definition';
+import { LogDriver, LogDriverConfig } from "../index";
+import { removeEmpty } from './utils';
+
+/**
+ * A log driver that sends logs to the specified driver.
+ */
+export class GenericLogDriver extends LogDriver {
+  /**
+   * The log driver to use for the container. The valid values listed for this parameter are log drivers
+   * that the Amazon ECS container agent can communicate with by default. You cannot use awslogs with the GenericLogDriver.
+   * You must use the AwsLogDriver if you want to use awslogs.
+   *
+   * For tasks using the Fargate launch type, the supported log drivers are awslogs and splunk.
+   * For tasks using the EC2 launch type, the supported log drivers are awslogs, syslog, gelf, fluentd, splunk, journald, and json-file.
+   *
+   */
+  private logDriver: string;
+
+  /**
+   * The configuration options to send to the log driver.
+   */
+  private options: { [key: string]: string };
+
+  /**
+   * Constructs a new instance of the GenericLogDriver class.
+   *
+   * @param props the generic log driver configuration options.
+   */
+  constructor(private readonly props: LogDriverConfig) {
```

Comment:
I'd really like this to take a `GenericLogDriverProps`. It might be that that happens to contain the exact same fields as a `LogDriverConfig`, but it is not the same: one is an input structure, the other one is an output structure. I'd prefer you copy/paste `LogDriverConfig` (so they can evolve independently if necessary).

Reply:
Agreed. Updated.